book/line_editor.md

```diff
@@ -1,124 +1,212 @@
-| --- | ------------------------------- |
-| d   | Delete                          |
-| p   | Paste after                     |
-| P   | Paste before                    |
-| h   | Move left                       |
-| l   | Move right                      |
-| j   | Move down                       |
-| k   | Move up                         |
-| w   | Move word right                 |
-| b   | Move word left                  |
-| i   | Enter Vi insert at current char |
-| a   | Enter Vi insert after char      |
-| 0   | Move to start of line           |
-| ^   | Move to start of line           |
-| $   | Move to end of line             |
-| u   | Undo                            |
-| c   | Change                          |
-| x   | Delete char                     |
-| s   | History search                  |
-| D   | Delete to end                   |
-| A   | Append to end                   |
-
-### Command History
+This can be changed at the commandline or persisted in `config.nu`.
+
+::: note
+Vi is a "modal" editor with "normal" mode and an "insert" mode. We recommend
+becoming familiar with these modes through the use of the Vim or Neovim editors
+before using Vi mode in Nushell. Each has a built-in tutorial covering the basics
+(and more) of modal editing.
+:::
+
+## Default Keybindings
+
+Each edit mode comes with common keybindings for Vi and Emacs text editing.
+
+### Emacs and Vi-insert Keybindings
+
+| Key                                        | Event                               |
+| ------------------------------------------ | ----------------------------------- |
+| <kbd>Shift</kbd>+<kbd>Enter</kbd>          | Insert newline                      |
+| <kbd>Alt</kbd>+<kbd>Enter</kbd>            | Insert newline                      |
+| <kbd>Backspace</kbd>                       | Backspace                           |
+| <kbd>End</kbd>                             | Move to end of line                 |
+| <kbd>End</kbd>                             | Complete history hint               |
+| <kbd>Home</kbd>                            | Move to line start                  |
+| <kbd>Ctrl</kbd>+<kbd>C</kbd>               | Cancel current line                 |
+| <kbd>Ctrl</kbd>+<kbd>L</kbd>               | Clear screen                        |
+| <kbd>Ctrl</kbd>+<kbd>R</kbd>               | Search history                      |
+| <kbd>Ctrl</kbd>+<kbd>→</kbd> (Right Arrow) | Complete history word               |
+| <kbd>Ctrl</kbd>+<kbd>→</kbd> (Right Arrow) | Move word right                     |
+| <kbd>Ctrl</kbd>+<kbd>←</kbd> (Left Arrow)  | Move word left                      |
+| <kbd>↑</kbd> (Up Arrow)                    | Move up                             |
+| <kbd>Ctrl</kbd>+<kbd>P</kbd>               | Move up                             |
+| <kbd>↑</kbd> (Up Arrow)                    | Move menu up                        |
+| <kbd>Ctrl</kbd>+<kbd>P</kbd>               | Move menu up                        |
```

Comment:
Not sure what the best presentation is for those modal/`UntilFound` bindings could be.

Reply:
That's a good point - They actually don't belong here at all since they don't apply to the Editing Mode.  These are *Menu* keybindings, so I've moved them to the Menus section of the chapter.  I also found (and fixed) a few more missing ones like <kbd>Tab</kbd>, <kbd>Shift</kbd>+<kbd>Tab</kbd>, and <kbd>Enter</kbd>.